platform-sdk/docs/proposals/README.md

```diff
@@ -0,0 +1,146 @@
+2. Core behaviors
+3. Is likely to performance
+4. Changes to interactions between major systems or teams (platform, services, mirror node, devops, etc)
+5. Module Structure & Repos
+
+---
+
+## Design Proposal Process Flow
+
+![](designProposalFlow.drawio.svg)
+
+## Creating A Proposal
+
+All proposals have an owner who is responsible for taking it through the process to completion. Proposals must be
+presented in the form of a PR containing the proposal in a subdirectory of `platform-sdk/docs/proposals/`. Proposals may
+contain diagrams and supporting materials. Diagrams must be created with DrawIO and exported in SVG format. All content
+must be editable (i.e. no images).
+
+If any APIs were developed in code as part of the design process, it may be included in the subdirectory of the proposal
+but not merged into the source code of the platform.
+
+While in this draft phase the proposal may be updated and modified as needed. It is best practice to solicit review from
+stakeholders to surface any potential change requests while the proposal is in the draft phase. All grammar and spelling
+errors should be caught during this phase.
+
+### Creating a Draft Proposal
+
+1. Create a new branch for the proposal.
+2. Create a subdirectory of `platform-sdk/docs/proposals/` with the name of the proposal.
+3. Copy the template `template.md` into the subdirectory and rename it to `README.md`
+4. Fill out the template with the details of the proposal, removing any irrelevant sections or adding new sections as
+   needed. Diagrams and other files may be included in the proposal directory and linked from the proposal.
+5. Create a draft PR with the proposal documents.
+6. Put the draft PR in the [Platform Design Proposals](https://github.com/orgs/hashgraph/projects/73/views/1) board in
+   the `Draft` status.
+7. Solicit feedback from stakeholders and other design collaborators.
+8. Iterate on the proposal until it is ready for voting. All open comments should be resolved while in the draft stage.
+
+### Submitting a Proposal for Voting
+
+When the proposal is ready for voting, take the following steps:
+
+1. Ensure all comments are resolved.
+2. Mark the PR as `Ready for Review`
+3. Update the status of the PR in
+   the [Platform Design Proposals](https://github.com/orgs/hashgraph/projects/73/views/1) board to `Voting`
+
+Requests for immaterial changes can be addressed by pushing additional commits to the existing proposal
+PR while in `Voting`. Examples of immaterial changes are spelling or grammar errors, wording changes that do not modify
+meaning or intent, and formatting changes. Examples of material changes include changes to behaviors or APIs, addition
+or removal of diagrams, etc. If any material changes are needed, the PR must be closed and moved to the `Superseded`
+status. A new proposal PR is prepared and voting is restarted. The `Superseded` PR must reference the new PR to
+create an auditable history.
```

Comment:
Is "avoiding a confusion about prior explicit votes" the only reason why a new PR needs to be open in this case, or there's any other important reasons that I may be missing?

As a software engineer, personally I'd prefer to continue iterating in the same PR given that GitHub explicitly resets prior approvals if a commit is pushed, and this helps eliminate the confusion automatically. Additionally, a diff history in the same PR is a lot more useful than a record of an "auditable history" proposed above because the latter cannot be used for partial, iterative diffs between the superseded and the new PRs, at least not as easily as if all the changes were a part of a single PR.

I wonder what do other people think?

Reply:
The burden of creating a whole new PR (copying then editing the old content) is minor enough to not slow progress, and an annoyance enough to cause the PR submitter to tune their internal sense of when a proposal is done.  Operant Conditioning with a mild negative stimulus to encourage more intelligent behavior.  It can become laborious if a developer makes it a habit of pushing premature proposals to a vote and simply iterates on the same PR until it is approved.  This discourages that behavior.

Reply:
>It can become laborious if a developer makes it a habit of pushing premature proposals to a vote and simply iterates on the same PR until it is approved.

@edward-swirldslabs you stated clearly what I was not able to. Thank you! I especially agree with this statement. There needs to be some expectation of effort and though put into a proposal before voting, because reviewing a proposal takes a lot of time. Every time a PR must be looked at again, it reduces the velocity of the teams as a whole. There should be some incentive not to push premature proposals.

Reply:
While I may dislike this policy personally because I don't see why updates are fully allowed in the Draft state and then disallowed in the Open state, and also because as the document states, future design proposal may in fact modify already approved ones - further diminishing the importance of disallowing updates in the Open state. But I cannot find any strong objective logical counter-arguments to what you said, and what you're saying sounds fair, I guess. So I'm resolving this conversation.